scipy/ndimage/_filters.py

```diff
@@ -1173,6 +1181,14 @@ def _min_or_max_filter(input, size, footprint, structure, output, mode,
         else:
             output[...] = input[...]
     else:
```

Comment:
To facilitate review, can you comment on what "separable" means and what is going on here? I could probably figure it out, but since one of us has already thought about it... I'd appreciate it! : )

Reply:
In the case of these 3 filters to get the equivalent of a 5 x 7 rectangular footprint with all values true, you could just apply a 1x7 footprint followed by a 5x1 footprint. In that sense, the footprint is separable and in some cases ndimage will detect this and run the separable case instead as it is typically more efficient. To avoid skipping the non-separable code path, I made sure to use a footprint that can't be split in this way.

The rank, percentile or median filters are not generally separable in the same way but the three filters in this PR are.

Reply:
Specifically, the following lines cause the footprint to not actually be used if it is true everywhere:
https://github.com/scipy/scipy/blob/26cd496418e4dcd694867333fce5ea0ab4c2fa8d/scipy/ndimage/_filters.py#L1136-L1141